gprofiler/spark/collector.py

```diff
@@ -10,28 +10,31 @@
 from urllib.parse import urljoin, urlparse
 
 import requests
+from bs4 import BeautifulSoup
 
 from gprofiler.log import get_logger_adapter
 from gprofiler.metrics import Sample
 from gprofiler.spark.metrics import (
+    SPARK_AGGREGATED_STAGE_METRICS,
     SPARK_APPLICATION_DIFF_METRICS,
     SPARK_APPLICATION_GAUGE_METRICS,
     SPARK_EXECUTORS_METRICS,
-    SPARK_RDD_METRICS,
-    SPARK_STAGE_METRICS,
+    SPARK_RUNNING_APPS_COUNT_METRIC,
     SPARK_STREAMING_BATCHES_METRICS,
     SPARK_STREAMING_STATISTICS_METRICS,
     SPARK_STRUCTURED_STREAMING_METRICS,
-    SPARK_TASK_SUMMARY_METRICS,
     YARN_CLUSTER_METRICS,
     YARN_NODES_METRICS,
 )
-from gprofiler.spark.mode import SPARK_DRIVER_MODE, SPARK_MESOS_MODE, SPARK_YARN_MODE
+from gprofiler.spark.mode import SPARK_MESOS_MODE, SPARK_STANDALONE_MODE, SPARK_YARN_MODE
 
 # Application type and states to collect
 YARN_SPARK_APPLICATION_SPECIFIER = "SPARK"
 YARN_RUNNING_APPLICATION_SPECIFIER = "RUNNING"
 
+SPARK_MASTER_STATE_PATH = "/json/"
```

Comment:
No need for trailing slash here. I see from the spark code that the json api is exposed at `{page}/json`.